test/components/form/list.spec.js

```diff
@@ -0,0 +1,28 @@
+import testData from '../../data';
+import { load } from '../../util/http';
+import { mockLogin } from '../../util/session';
+
+describe('FormList', () => {
+  describe('create button', () => {
+    it('shows the button to an administrator', async () => {
+      mockLogin({ role: 'admin' });
+      testData.extendedProjects.createPast(1);
+      const app = await load('/projects/1');
+      app.first('#form-list-create-button').should.be.visible();
+    });
+
+    it('does not render the button for a project viewer', async () => {
```

Comment:
Add test also for data collector? I'm not sure -- the combination of all possible roles and features is getting pretty large.

Reply:
I was thinking something similar. Since the logic is so simple (it just checks a single verb, `form.create`), I think it's probably enough just to test the two cases here, a role with `form.create` and a role without `form.create`. However, in other places where more complex logic is in place, I've tended toward testing more roles.